Change IAM docs to focus on IAM v2 (#2715)
chef/automate: components/automate-chef-io/content/docs/api-tokens.md

    weight = 70
+++

## Overview

> Chef Automate API Tokens allow access to the Chef Automate API.

Author: ```suggestion
API tokens grant authorized access to the Chef Automate API.
```

> Permission on the `iam:tokens` resource is required to interact with tokens. Any user that is part of the `admins` team, or the `Administrator` policy will have this permission. Otherwise, [IAM v2 custom policies]({{< relref "iam-v2-guide.md#creating-custom-policies" >}}) can be created to assign this permission.

Reviewer: ```suggestion
Permission on the `iam:tokens` resource is required to interact with tokens. Any user that is part of the `admins` team or the `Administrator` policy will have this permission. Otherwise, [IAM v2 custom policies]({{< relref "iam-v2-guide.md#creating-custom-policies" >}}) can be created to assign this permission.
```

## Managing API Tokens

### Creating API Tokens

> Navigate to _API Tokens_ in the **Settings** tab. Then use the **Create Token** button, which opens a helper window for entering the token's _name_ and a unique _ID_, and optionally assigning the API token to some _Projects_. Token IDs can contain lower-case letters, hyphens, underscores, and numbers, but they cannot contain symbols or upper-case letters.

Author: 1. so helper window is what was there before these edits, and i was trying not to change stuff. but i think ill leave that for @mjingle to decide.

> Navigate to _API Tokens_ in the **Settings** tab. Then use the **Create Token** button, which opens a helper window for entering the token's _name_ and a unique _ID_, and optionally assigning the API token to some _Projects_. Token IDs can contain lower-case letters, hyphens, underscores, and numbers, but they cannot contain symbols or upper-case letters.

Author: oh 2. was there as well, and i have a faint memory that was a docs team edit so will leave as is until their portion of the work.

> Navigate to _API Tokens_ in the **Settings** tab. Then use the **Create Token** button, which opens a helper window for entering the token's _name_ and a unique _ID_, and optionally assigning the API token to some _Projects_. Token IDs can contain lower-case letters, hyphens, underscores, and numbers, but they cannot contain symbols or upper-case letters.

Author: ive addressed 2., that content is no longer there now.

> Navigate to _API Tokens_ in the **Settings** tab. Then use the **Create Token** button, which opens a helper window for entering the token's _name_ and a unique _ID_, and optionally assigning the API token to some _Projects_. Token IDs can contain lower-case letters, hyphens, underscores, and numbers, but they cannot contain symbols or upper-case letters.

Reviewer: Changes were made to use "dialog box" instead of "helper window."

> After a token is created from the UI it will have no permissions. To assign the token permissions, navigate to _Policies_ in the **Settings** tab, locate the appropriate policy, and then add the token as a member of the policy using a [member expression]({{< relref "iam-v2-guide.md#member-expressions" >}}).

Reviewer: ```suggestion
After a token is created from the UI, it will have no permissions. To assign the token permissions, navigate to _Policies_ in the **Settings** tab, locate the appropriate policy, and then add the token as a member of the policy using a [member expression]({{< relref "iam-v2-guide.md#member-expressions" >}}).
```

> After a token is created from the UI, it will have no permissions. To assign the token permissions, navigate to _Policies_ in the **Settings** tab, locate the appropriate policy, and then add the token as a member of the policy using a [member expression]({{< relref "iam-v2-guide.md#member-expressions" >}}).

Author: do you think this is unclear though in relation to admin tokens? i guess thats what i was trying to distinguish between

> After a token is created from the UI, it will have no permissions. To assign the token permissions, navigate to _Policies_ in the **Settings** tab, locate the appropriate policy, and then add the token as a member of the policy using a [member expression]({{< relref "iam-v2-guide.md#member-expressions" >}}).

Reviewer 2: Only in the sense that less words conveying the same message is better, and that better to avoid passive voice.

> After a token is created from the UI, it will have no permissions. To assign the token permissions, navigate to _Policies_ in the **Settings** tab, locate the appropriate policy, and then add the token as a member of the policy using a [member expression]({{< relref "iam-v2-guide.md#member-expressions" >}}).

Author: ill leave this comment for @mjingle as well then.

> After a token is created from the UI, it will have no permissions. To assign the token permissions, navigate to _Policies_ in the **Settings** tab, locate the appropriate policy, and then add the token as a member of the policy using a [member expression]({{< relref "iam-v2-guide.md#member-expressions" >}}).

Reviewer: Addressed.

![API Tokens](/images/docs/admin-tab-API-tokens-list.png)

#### API Token Value
After creating an API Token, you can obtain the token's value by opening the menu at the end of the table row and selecting **Copy Token**.

#### Admin Tokens

> Admin tokens are tokens that are automatically added to the Administrator policy, which grants full access to Chef Automate. Admin tokens can only be created using the `chef-automate` command line.

Author: i kinda wanted to leave that to the documentation on policies.. and keep this specific to the tokens page in the ui. once we add policies to the tokens page then i think we can add that kind of content.

    weight = 70
+++

## Overview
Chef Automate API Tokens allow access to the Chef Automate API.

> Permission on the `iam:tokens` resource is required to interact with tokens. Any user that is part of the `admins` team, or the `Administrator` policy will have this permission, otherwise, [IAM v2 custom policies]({{< relref "iam-v2-guide.md#creating-custom-policies" >}}) can be created to assign this permission.

Reviewer 3: ```suggestion
Permission on the `iam:tokens` resource is required to interact with tokens. Any user that is part of the `admins` team, or the `Administrator` policy will have this permission. Otherwise, [IAM v2 custom policies]({{< relref "iam-v2-guide.md#creating-custom-policies" >}}) can be created to assign this permission.
```

Reviewer 3: Side note: Why is `Administrator` single and `Editors/Viewers` plural? 😩

## Managing API Tokens

### Creating API Tokens

> Navigate to _API Tokens_ in the **Settings** tab. Then use the **Create Token** button, which opens a helper window for entering the token's _name_, a unique _ID_, and optionally assign the API token to some _Projects_. Token IDs can contain lower-case letters, hyphens, underscores, and numbers, but they cannot contain symbols or upper-case letters.

Reviewer 3: ```suggestion
Navigate to _API Tokens_ in the **Settings** tab. Then use the **Create Token** button, which opens a helper window for entering the token's _name_ and a unique _ID_, and optionally assigning the API token to some _Projects_. Token IDs can contain lower-case letters, hyphens, underscores, and numbers, but they cannot contain symbols or upper-case letters.
```

> After a token is created from the UI it will have no permissions, to assign the token permissions navigate to _Policies_ in the **Settings** tab, locate the appropriate policy and then add the token as a member of the policy using a [member expression]({{< relref "iam-v2-guide.md#member-expressions" >}}).

Reviewer 3: ```suggestion
After a token is created from the UI it will have no permissions. To assign the token permissions, navigate to _Policies_ in the **Settings** tab, locate the appropriate policy, and then add the token as a member of the policy using a [member expression]({{< relref "iam-v2-guide.md#member-expressions" >}}).
```

![API Tokens](/images/docs/admin-tab-API-tokens-list.png)

#### API Token Value

> After creating an API Token you can obtain the token's value by opening the menu at the end of the table row and selecting **Copy Token**.

Reviewer 3: ```suggestion

After creating an API Token, you can obtain the token's value by opening the menu at the end of the table row and selecting **Copy Token**.
```

#### Admin Tokens

> Admin tokens are tokens that are automatically added to the admin policy which grants full access to Chef Automate. Admin tokens can only be created using the `chef-automate` command line.

Reviewer 3: ```suggestion
Admin tokens are tokens that are automatically added to the Administrator policy, which grants full access to Chef Automate. Admin tokens can only be created using the `chef-automate` command line.
```

```
chef-automate iam token create <your-token-name> --admin
```

To create an admin token and immediately store it in an environment variable for
easy access, you can instead run:

```bash
export TOKEN=`chef-automate iam token create <your-token-name> --admin`
echo $TOKEN
```

Once you have an Admin API token, you can use it to make requests by passing it in the `api-token`
header:

```bash
curl -s -H "api-token: $TOKEN" https://{{< example_fqdn "automate" >}}/apis/iam/v2/policies -v
```

### Deleting API Tokens
Navigate to _API Tokens_ in the **Settings** tab. Then open the menu at the end of the table row and select **Delete Token**.

### Changing API Token Details

> The API token name, projects a team belongs to, and the token's status can be changed by navigating to _API Tokens_ from the **Settings** tab, selecting an individual token and then navigating to the **Details** tab.

Author: ```suggestion
The API token name, projects a token belongs to, and the token's status can be changed by navigating to _API Tokens_ from the **Settings** tab, selecting an individual token and then navigating to the **Details** tab.
```
